### PR TITLE
Python PIP test-on-install

### PR DESCRIPTION
**Pull request by caisq**
PIP package is built and installed in a Docker container. Then the Python unit tests in the TF source are moved to a separate folder and ran with the plain Python environment without Blaze.

Four out of 151 tests were skipped for now due to dependency on modules in TF source that are not publicly exported.

Tests performed: Jenkins experimental builds for CPU and GPU Python test-on-install, see passing results at: 
http://ci.tensorflow.org/view/Experimental/job/experimental-cais-tensorflow-cpu-python27-copt_pip_install-test/1/consoleFull
http://ci.tensorflow.org/view/Experimental/job/experimental-cais-tensorflow-gpu-python27-copt_pip_install-test/1/consoleFull

### CODE REVIEW

**Comment by martinwicke** [comment]
Can you either sign the CLA for your other email address or amend the
commits (or squash them) to use only your CLA-ok email?

**Comment by caisq** [comment]
HI Martin,

I thought the email (cais@google.com, the only Google email I have) should
have been registered. Anyway, if necessary, I can fill out the CLA. What is
the "google group email address" I should put in the form?

Thanks,
Shanqing

On Fri, Jan 29, 2016 at 7:08 PM, Martin Wicke <notifications@github.com>
wrote:

> Can you either sign the CLA for your other email address or amend the
> commits (or squash them) to use only your CLA-ok email?
> On Fri, Jan 29, 2016 at 08:30 caisq <notifications@github.com> wrote:
>
> > PIP package is built and installed in a Docker container. Then the Python
> > unit tests in the TF source are moved to a separate folder and ran with
> the
> > plain Python environment without Blaze.
> >
> > Four out of 151 tests were skipped for now due to dependency on modules
> in
> > TF source that are not publicly exported.
> >
> > Tests performed: Jenkins experimental builds for CPU and GPU Python
> > test-on-install, see passing results at:
> >
> >
> http://ci.tensorflow.org/view/Experimental/job/experimental-cais-tensorflow-cpu-python27-copt_pip_install-test/1/consoleFull
> >
> >
> http://ci.tensorflow.org/view/Experimental/job/experimental-cais-tensorflow-gpu-python27-copt_pip_install-test/1/consoleFull
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/tensorflow/tensorflow/pull/935
> > Commit Summary
> >
> > - Python PIP test on install
> > - Improving test_on_install.sh
> > - Minor topy fix
> >
> > File Changes
> >
> > - *M* tensorflow/tools/ci_build/Dockerfile.cpu
> > <https://github.com/tensorflow/tensorflow/pull/935/files#diff-0> (8)
> > - *M* tensorflow/tools/ci_build/Dockerfile.gpu
> > <https://github.com/tensorflow/tensorflow/pull/935/files#diff-1> (9)
> > - *A* tensorflow/tools/ci_build/builds/test_on_install.sh
> > <https://github.com/tensorflow/tensorflow/pull/935/files#diff-2> (120)
> > - *A* tensorflow/tools/ci_build/ci_test_on_install.sh
> > <https://github.com/tensorflow/tensorflow/pull/935/files#diff-3> (81)
> >
> > Patch Links:
> >
> > - https://github.com/tensorflow/tensorflow/pull/935.patch
> > - https://github.com/tensorflow/tensorflow/pull/935.diff
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/tensorflow/tensorflow/pull/935>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/935#issuecomment-177026119>
> .
>



-- 
---
Shanqing Cai
Software Engineer, Tools and Infrastructure
Applied Machine Intelligence
Google, Inc.
cais@google.com

**Comment by martinwicke** [comment]
One of the commits is created using "cais <cais@ci.tensorflow.org>", maybe just squash the commits, then you should be fine.

File: tensorflow/tools/ci_build/builds/test_on_install.sh [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#
+# To select only a subset of the Python tests to run, set the environment
+# variable PY_TEST_WHITELIST, e.g.,
+#   PY_TEST_WHITELIST="tensorflow/python/kernel_tests/shape_ops_test.py"
+#
+# Seprate the tests with a space. Leave this environment variable empty to
+# disable the whitelist.
+
+# You can also ignore a set of the tests by using the variable PY_TEST_BLACKLIST
+# These are the test that depend on Python modules in the source that are not
```
Comment (martinwicke) [inline tensorflow/tools/ci_build/builds/test_on_install.sh]: Add documentation for PY_TEST_GPU_BLACKLIST

**Comment by caisq** [comment]
I'm closing this pull request and creating a new one to address wicke's comments and the cla issue. See https://github.com/tensorflow/tensorflow/pull/939

**Pull request closed without merging**